api/README.md

```diff
@@ -0,0 +1,32 @@
+# VS Code clangd Extension API
+
+The VS Code clangd extension exposes an API that other extensions can consume:
+
+```typescript
+import * as vscode from 'vscode';
+import type { ClangdExtension, ASTParams, ASTNode } from '@clangd/vscode-clangd';
+
+const CLANGD_EXTENSION = 'llvm-vs-code-extensions.vscode-clangd';
+const CLANGD_API_VERSION = 1;
+
+const ASTType = 'textDocument/ast';
+
+const provideHover = (document: vscode.TextDocument, position: vscode.Position, _token: vscode.CancellationToken): Promise<vscode.Hover | undefined> => {
+
+    const clangdExtension = vscode.extensions.getExtension<ClangdExtension>(CLANGD_EXTENSION);
+
+    if (clangdExtension) {
+        const api = (await clangdExtension.activate()).getApi(CLANGD_API_VERSION);
```

Comment:
We should probably discourage consumers unconditionally calling `activate()`, since the clangd extension may already be activated and it doesn't expect `activate()` to be called again in that state.

I believe the right pattern here is for consuming code to:
 * use `clangdExtension.exports` to access the `ClangdExtension` object
 * if it's unsure whether the clangd extension is active yet at that point, then guard the access to `.exports` with a `clangdExtension.isActive` check
    * if the check fails, either bail or call `activate()` at that point

This is what I've done in my [example plugin](https://github.com/HighCommander4/vscode-clangd-api-example/blob/5007b4a1b0252f50308cc9b8cdcda4e6d9d8992e/src/extension.ts#L12-L19), and I'd prefer the README does the same. (Sorry to be picky on this point, I worry that users who may not have much familiarity with VSCode extension may look to this README for how to use the API and assume that what it shows is a good way to do things.)

Reply:
`vscode.Extension.activate()` is designed to be called multiple times and only calls into the extension's `activate()` method once. Subsequent calls return the resolved extension. This is a common pattern and (just to be sure) I tested this scenario by activating it multiple times with logging in the extension `activate()` function. It worked as expected and only activates once.

Reply:
Thanks for checking on this, I think this is fine then.